explainers/hidden-content-explainer.md

```diff
@@ -135,56 +128,45 @@ sections already has an event handler to expand and show the section. With the
-       based on scroll offsets and intersection observations.
-2. Scroll to text navigation. Depending on the implementation of scroll-to-text,
-   from the page's perspective the behavior is indistinguishable from
-   find-in-page behavior. This means that it reveals where the searched text is
-   located but not what it is. Furthemore, this information is only available to
-   the page using the `beforematch` event, and is not observable by any third
-   parties.
-
-Some consideration needs to be given to the `beforematch` event when it comes to
-cross frame interactions. Specifically, if the matched information is found within
-a frame, should the signal propagate to the embedding page?
-
-In general, we believe that the `beforematch` event does not pose a privacy problem.
-However, the privacy aspect of the event should be discussed in more detail via
-formal security reviews.
+The beforematch event could expose more information to the page than is
+currently exposed because it very directly hints to the text the user is
+searching for since it gets fired on the element containing the target text.
+
+Note that it is already possible to snoop on find-in-page by creating a very
+small scrollable area containing every next character the user could type into
+find-in-page, listening to scroll events to see which character the user typed
+in, then prepending the new character to all of the next possible search terms.
+This is demonstrated in
+[search-incremental.html](/resources/find-in-page/search-incremental.html).
+
+In order to mitigate the additional information that beforematch could expose to
+the page, we have added two constraints to when beforematch can be fired:
+
+1. The beforematch event will only be fired on `content-visibility:
+hidden-matchable` content. This removes the information delta
+of find-in-page on normal pages without hidden find-in-page snooping. Without
+constraint, a page could listen to beforematch events and get some
+idea of what you’re searching for within the text already in the page. Although
+a page could already try to do this by listening to scroll events and guessing
+which element was scrolled to, the beforematch event is a bit of a stronger and
+more accurate signal in this case. By firing beforematch only on
+hidden-matchable content, we eliminate this possibility.
+
+2. If the page doesn’t reveal hidden-matchable content in response to a
+beforematch event, the beforematch event will not be fired for the remainder of
+the lifetime of the document. This will prevent the page from building out a
+string of what the user is searching for without having to reveal that string
+to the user visually. Although this is already possible as shown in
+[search-incremental.html](/resources/find-in-page/search-incremental.html),
+firing beforematch on hidden content that stays hidden would be harder for the
+user to notice.
+
+As for the ScrollToTextFragment case, the page can already guess with reasonable
```

Comment:
FWIW ScrollToTextFragment also just happens once on the page, so we also only fire the event in the hidden-matchable subtree. An interesting question is what happens if the page doesn't reveal it: does it mean that find-in-page also won't fire the event because of restriction 2? Or is scroll to text fragment's beforematch event logically different from find-in-page beforematch event?

Reply:
I clarified this, how does it look?

Reply:
Looks good